java/util/Date/Date.hpp

```diff
@@ -24,365 +24,566 @@
+             */
+            void update() {
+                // Update changes
+                this->timer = mktime(this->localTimer);
+
+                this->sec = this->localTimer->tm_sec;
+                this->min = this->localTimer->tm_min;
+                this->hour = this->localTimer->tm_hour;
+                this->mday = this->localTimer->tm_mday;
+                this->mon = this->localTimer->tm_mon;
+                this->year = this->localTimer->tm_year;
+                this->wday = this->localTimer->tm_wday;
+                this->yday = this->localTimer->tm_yday;
+                this->isdst = this->localTimer->tm_isdst;
+                this->gmtoff = this->localTimer->tm_gmtoff;
+                this->zone = this->localTimer->tm_zone;
+
+                this->defaultCenturyStart = (this->year / 100) * 100;
+            }
+
+            /**
+             * Allocates a Date object and initializes it so that
+             * it represents the instant at the start of the second specified
+             * by the year, month, date,
+             * hrs, min, and sec arguments,
+             * in the local time zone.
+             *
+             * @param   year    the year minus 1900.
+             * @param   month   the month between 0-11.
+             * @param   date    the day of the month between 1-31.
+             * @param   hrs     the hours between 0-23.
+             * @param   min     the minutes between 0-59.
+             * @param   sec     the seconds between 0-59.
+             */
+            void initialize(int year, int month, int date,
+                            int hrs, int min, int sec) {
+                tm localTimer = { 0 };
+
+                localTimer.tm_year = year % 1900;
+                localTimer.tm_mon = month;
+                localTimer.tm_mday = date;
+                localTimer.tm_hour = hrs;
+                localTimer.tm_min = min;
+                localTimer.tm_sec = sec;
+
+                this->timer = mktime(&localTimer);
+
+                this->localTimer = localtime(&this->timer);
+
+                update();
+            }
+
+            /**
+             * Allocates a Date object and initializes it
+             *
+             * @param timer time milliseconds after
+             *              January 1, 1970 00:00:00 GMT.
+             */
+            void initialize(time_t timer) {
+                this->timer = timer;
+                this->localTimer = localtime(&this->timer);
+
+                update();
+            }
+
+            /**
+             * Convert the time milliseconds after
+             * January 1, 1970 00:00:00 GMT to string
+             * type - represent the Date object
+             *
+             * @param timePresenter
+             * @return String
+             */
+            string timeToString(string pattern, tm *timeManagement) {
+                size_t size = 100;
+                auto result = static_cast<string> (malloc(size * sizeof(char)));
```

Comment:
Why `auto` when you already static_cast<string>.

Reply:
Please remove `auto` and static_cast, use `calloc` instead of `malloc` for safety.

Reply:
yep, thank you. I have changed it into calloc.